source/modulo_core/src/Cell.cpp

```diff
@@ -16,18 +16,25 @@ Cell::Cell(const rclcpp::NodeOptions& options) :
     shutdown_(false),
     period_(utilities::parse_period(options)) {
   // add the update parameters call
-  this->update_parameters_timer_ = this->create_wall_timer(this->period_, [this] { this->update_parameters(); });
+  this->add_daemon([this] { this->update_parameters(); }, this->period_);
 }
 
 Cell::~Cell() {
-  this->parameters_.clear();
   this->on_shutdown();
 }
 
 void Cell::reset() {
   this->active_ = false;
   this->configured_ = false;
-  this->handlers_.clear();
```

Comment:
This makes it evident that handlers cannot be added in the constructor, since this would remove the interfaces when unconfiguring and reconfiguring a node. Regular handlers, even if they are "always active", will never work in an unconfigured state, since they must be added "on_configure".

Since Component predicate functions are now daemonized, they persist throughout the cleanup and reset. It makes sense now!

Reply:
Yes exactly. Now with this PR you can add handlers in the constructor as well as long as you declare them `always_active`. This is actually how it is done in the `Component` class for predicates.

Reply:
Ah yes perfect. In fact I wrote it wrong in my first comment which is how it was before this PR. Since now the reset() only erases non-always active handlers, it looks like it works as intended!